### ISSUE TITLE
Strip leading and trailing spaces from most/all fields

### ISSUE THREAD

**Issue by zorae** [internetarchive/openlibrary#10130]
### Proposal

When editing an author, work, or edition, invisible characters such as spaces and tabs should be automatically stripped from the beginning and end of a field.

### Justification

Problem: Librarians may accidentally include leading/trailing spaces, which – if not manually removed – are saved to the database. This can lead to e.g. author names looking weird.

Impact: Librarians can save time, e.g. when pasting names from external sites.

Note: We would have to check for edge cases in which leading/trailing spaces have a legitimate use and exclude those fields.

#### Requirements Checklist

* [ ]

#### Related files

*

#### Stakeholders

*

<hr>

#### Instructions for Contributors

Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by QuangPhung15** [comment]
Hi @zorae,

I want to work on this issue. It is a useful improvement to streamline data entry and prevent formatting errors. I'll review potential edge cases where leading/trailing spaces might be necessary and ensure the implementation excludes those fields.

Please let me know if this feature has any additional details or preferences!

**Comment by zorae** [comment]
Thanks @QuangPhung15!
I’ve been pondering if spaces should a) be stripped immediately upon deselecting a field, or b) only upon saving changes. 

a) has the advantage that it gives librarians immediate feedback so they know they don‘t have to manually remove spaces. But I foresee that this could be annoying for multi-line fields like bio or work description where a librarian might start writing something, add a space or line break at the end and deselect the field before coming back to continue writing.

Option a) would have to be implemented client-side, while b) could be either client- or server-side.

**Comment by bitnapper** [comment]
Would it also be possible to substitute multiple space characters to only one, something like: `s/\S+/ /g`?

**Comment by mekarpeles** [comment]
@zorae can we please be enumerative of the exact fields you believe this should be applied to? Even if we start with only a few, that is more likely to make forward progress than requiring folks who don't have librarian context / experience from having to figure out what fields we might be talking about. Thank you!

**Comment by zorae** [comment]
@mekarpeles Some fields I would find particularly useful are author name, bio, date of birth, date of death, edition title, work title, and work description.

Can you clarify what the issue would be with the “, Sr.” suffix? It ends with a period, so I don’t see how a rule that strips out whitespace at the end of a field would affect it.

**Comment by scottbarnes** [comment]
I apologize, I think I misheard this when the issue came up as including the stripping of periods.

**Comment by Spaarsh** [comment]
@zorae the first option seems far better to be since it would immediately make the librarians aware of this new update. 

However, I do not feel that any of this should need server-side processing. 

If there is a clear consensus about this, then I would like to begin working on this rightaway since it seems simple enough to implement by adding the relevant eventHandlers for [edit.html](https://github.com/internetarchive/openlibrary/blob/f3b26c2c0721f8713353fe4b341230332e30008d/openlibrary/templates/books/edit.html#L4).

**Comment by scottbarnes** [comment]
Let's try with this happening upon de-selection of the relevant fields that @zorae enumerated. I went ahead and assigned this to you, @Spaarsh.

**Comment by Spaarsh** [comment]
@scottbarnes I have added the relevant JS and these are the results:
For editing an existing edition:
<video src="https://github.com/user-attachments/assets/39927479-92fb-4a74-86bc-ba94219e593e" width="320" height="240" controls></video>

For adding a new edition:
<video src="https://github.com/user-attachments/assets/9272c1c0-97e5-4c25-898f-2e9a07cbd281" width="320" height="240" controls></video>

For adding a new book:
<video src="https://github.com/user-attachments/assets/6d4188f5-7f9a-49f2-b279-a8ec7904a9a8" width="320" height="240" controls></video>

**Comment by zorae** [comment]
@Spaarsh looks great! does this handle only normal spaces, or also other whitespace characters like tabs?

**Comment by Spaarsh** [comment]
@zorae yes! I have used the JavaScript method [trim()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim) and the official document defines whitespaces [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#white_space) and it includes tabs as well. It also removes [line-terminators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#line_terminators) such as /n.
